packages/next/build/analysis/get-page-static-info.ts

```diff
@@ -158,6 +159,19 @@ function getMiddlewareConfig(
     result.matchers = getMiddlewareMatchers(config.matcher, nextConfig)
   }
 
+  if (config.allowDynamic) {
+    result.allowDynamicGlobs = Array.isArray(config.allowDynamic)
+      ? config.allowDynamic
+      : [config.allowDynamic]
+    for (const glob of result.allowDynamicGlobs ?? []) {
+      if (typeof glob !== 'string') {
+        throw new Error(
+          `A middleware/edge exported 'config.allowDynamic' must be a string or an array of strings`
+        )
+      }
```

Comment:
suggestion: how about validating glob patterns to show better error message like `invalid glob pattern in config.allocDynamic: ...`? For example, it appears the matcher library [might throw if the pattern is empty](https://github.com/micromatch/picomatch/blob/13efdf0c7d0e07ee30bf78d0079184aa4a0ec7d4/lib/picomatch.js#L48).

Reply:
Done. Thanks for this good idea!!